src/main/java/net/dries007/tfc/world/classic/biomes/BiomeDecoratorTFC.java

```diff
@@ -42,14 +61,76 @@ public BiomeDecoratorTFC(int lilyPadPerChunk, int waterPlantsPerChunk)
         this.mushroomBrownGen = null;
         this.mushroomRedGen = null;
         this.bigMushroomGen = null;
-        this.reedGen = null;
 
         plantGen = new WorldGenPlantTFC();
 
-        reedGen = new WorldGenTallPlant(Blocks.REEDS); // todo: replace block?
         sandGen = new WorldGenSandTFC(7);
         pumpkinGen = new WorldGenPumpkinTFC(Blocks.PUMPKIN); // todo: replace block?
-        waterplantGen = new WorldGenWaterPlants(); // todo: replace block
+
+        for (Plant plant : TFCRegistries.PLANTS.getValuesCollection())
+        {
+            switch (plant.getPlantType().toString())
```

Comment:
Two things:
1) Don't switch on Enum#toString(), it just makes code messier to clean up later. And,
2) See the earlier suggestion about adding a `PlantType#getEnumType()` to clean this up a bit. Maybe a `int[] plantCountsByEnumPlantTypeIndex`